src/request.rs

```diff
@@ -4,19 +4,34 @@ use serde::{de::DeserializeOwned, Serialize};
 use serde_json::{from_str, to_string};
 
 #[derive(Debug)]
-pub(crate) enum Method<T: Serialize> {
-    Get(T),
-    Post(T),
-    Patch(T),
-    Put(T),
-    Delete,
+pub(crate) enum Method<Q, B> {
+    Get { query: Q },
+    Post { query: Q, body: B },
+    Patch { query: Q, body: B },
+    Put { query: Q, body: B },
+    Delete { query: Q },
 }
```

Comment:
It's really nice that you took the time to convert every variant to the struct format :+1: